docs/06-actors-actions.md

```diff
@@ -0,0 +1,398 @@
+
+```ts
+/**
+ * DANGER: This is for advanced users to totally override draw logic.
+ */
+public update(ctx: CanvasRenderingContext2D, delta: number) {
+
+  // Invoke the core logic
+  // Or leave it out to completely override
+  super.update(engine, delta);
+
+  // Perform custom update logic
+  // ...
+}
+```
+
+### Post-update
+
+[[Actor.onPostUpdate]] is called after [[Actor.update]] to prepare state for the _next_ frame.
+
+<docs-note>**Important:** At this time, the frame hasn't been drawn yet so state updated in this method will be reflected during the draw loop but will **not** be reflected in the scene graph until the _next_ frame.</docs-note>
+
+```ts
+class Player extends Actor {
+  /**
+   * Runs after "core" update logic, before the next frame
+   */
+  public onPostUpdate(engine: ex.Engine, delta: number) {
+    // check if player died
+    if (this.health <= 0) {
+      this.kill()
+      return
+    }
+  }
+}
+```
+
+## Drawing actors
+
+Actors by default have no associated [drawings](/docs/drawings), meaning that they will be rendered without any graphics unless you've assigned a default [[Actor.color]] or attached a drawing. If an actor has a color set, it will draw a box in that color. This is useful only at the beginning of development when you're just tinkering but for most games you'll need to add sprites, animations, and other drawings.
+
+### Working with Textures & Sprites
+
+Think of a [[Texture|texture]] as the raw image file that will be loaded into Excalibur. In order for it to be drawn
+it must be converted to a [[Sprite]].
+
+A common usage is to load a [[Texture]] and convert it to a [[Sprite]] for an actor. If you are using the [[Loader]] to
+pre-load assets, you can simply assign an actor a [[Sprite]] to draw. You can also create a
+[[Texture.asSprite|sprite from a Texture]] to quickly create a [[Sprite]] instance.
+
+```ts
+// assume Resources.TxPlayer is a 80x80 png image
+
+public onInitialize(engine: ex.Engine) {
+
+   // set as the "default" drawing
+   this.addDrawing(Resources.TxPlayer);
+
+   // you can also set a Sprite instance to draw
+   this.addDrawing(Resources.TxPlayer.asSprite());
+}
+```
+
+### Working with Animations
+
+A [[SpriteSheet]] holds a collection of sprites from a single [[Texture]].
+Use [[SpriteSheet.getAnimationForAll]] to easily generate an [[Animation]].
+
+```ts
+// assume Resources.TxPlayerIdle is a texture containing several frames of an animation
+
+public onInitialize(engine: ex.Engine) {
+
+   // create a SpriteSheet for the animation
+   const playerIdleSheet = new ex.SpriteSheet(Resources.TxPlayerIdle, 5, 1, 80, 80);
+
+   // create an animation
+   const playerIdleAnimation = playerIdleSheet.getAnimationForAll(engine, 120);
+
+   // the first drawing is always the current
+   this.addDrawing("idle", playerIdleAnimation);
+}
+```
+
+### Custom drawing
```

Comment:
@eonarheim Again here too, did I get the details right? I would like to include a diagram for both the update and draw loop hooks.